### PR TITLE
RN/Relay: Bump Dependency to React 19 (#44990)

### PR DESCRIPTION
**Pull request by blakef**
Summary:
Pull Request resolved: https://github.com/facebook/react-native/pull/44990

Upgrades React Native and Relay to depend on React 19, which is currently published as release candidates. This is in preparation for React Native 0.75.

This will depend on updating open source renderers after [facebook/react#29903](https://github.com/facebook/react/pull/29903) is merged.

Changelog:
[General][Changed] - Upgrade to React 19

Reviewed By: robhogan

Differential Revision: D58625271

fbshipit-source-id: f9ad95b18716a9ce02f7cfbcc7248bdfb244c010

Original: https://github.com/facebook/react-native/commit/0dd8224a711dfa9016a200a6d489020e46304d07

### CODE REVIEW

**Review by cortinico — APPROVED** [review]

File: template/template/android/app/build.gradle [hunk 1]
```diff
@@ -49,6 +49,9 @@ react {
     //
     //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
     // hermesFlags = ["-O", "-output-source-map"]
+
+    /* Autolinking */
+    autolinkLibrariesWithApp()
```
Comment (cortinico) [inline template/template/android/app/build.gradle:54]: nit: Those changes are unrelated from the linked PR @blakef (https://github.com/facebook/react-native/pull/44990)

Comment (blakef) [reply template/template/android/app/build.gradle:54]: Yea, my script has failed me here.
